test/e2e/nodemaintenance_quorumsize_test.go

```diff
@@ -0,0 +1,211 @@
+package e2e
+
+import (
+	goctx "context"
+	"fmt"
+	"testing"
+	"time"
+	operator "kubevirt.io/node-maintenance-operator/pkg/apis/nodemaintenance/v1beta1"
+	nmo "kubevirt.io/node-maintenance-operator/pkg/controller/nodemaintenance"
+	nmooperator "kubevirt.io/node-maintenance-operator/pkg/controller/nodemaintenance"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	policy "k8s.io/api/policy/v1beta1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func createEtcPDB(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) (*policy.PodDisruptionBudget, *corev1.Namespace, error) {
+
+	nsobject := &corev1.Namespace{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Namespace",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nmooperator.OpenshiftMachineConfigNamespace,
+			Labels: map[string]string{
+				"name":	nmooperator.OpenshiftMachineConfigNamespace,
+			},
+		},
+	}
+	err := f.Client.Create(goctx.TODO(), nsobject, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	if err != nil {
+		return nil, nil, fmt.Errorf("can't create namespace : %v", err)
+	}
+
+	objectName  := "etcpdb"
+	minAvailable := intstr.FromInt(1)
+	object := &policy.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      objectName,
+			Namespace: nmooperator.OpenshiftMachineConfigNamespace,
+		},
+		Spec: policy.PodDisruptionBudgetSpec{
+			Selector:     &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"k8s-app": "etcd-quorum-guard",
+				},
+			},
+			MinAvailable: &minAvailable,
+		},
+		Status: policy.PodDisruptionBudgetStatus{},
+	}
+
+	err = f.Client.Create(goctx.TODO(), object, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	if err != nil {
+		f.Client.Delete(goctx.TODO(), nsobject)
+		return nil, nil, err
+	}
+	return object, nsobject, nil
+}
+
+func checkQuorumSizeViolation(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+
+	pdb, nsobject, err := createEtcPDB(t , f , ctx )
+	if err != nil {
+		showDeploymentStatus(t, f, fmt.Errorf("can't create test pdb :  %v", err))
```

Comment:
This isn't an error if the PDB already exists, which is what we expect on an OCP 4 cluster

Reply:
we create the pdb it in all the test environments (including ocp 4.3) If that weren't the case then createEtcPDB would have failed at line #59 with error.IsAlreadyExist.

Reply:
CI isn't the only place these tests can/should/will run, and kubevirtci isn't the only CI that could be used.
Marc already has a todo item for getting access to the OCP CI

Reply:
Makes sense. Added 1f312fe that deletes the objects if they were created for the purpose of this test.